zhaquirks/ctm/__init__.py

```diff
@@ -0,0 +1,264 @@
+    """Active status of the stove guard (In use)."""
+
+    Inactive = 0x00
+    Active = 0x01
+
+
+class CTMDiagnosticsCluster(CustomCluster):
+    """CTM Lyng custom diagnostics cluster."""
+
+    name = "CTMDiagnostics"
+    cluster_id = 0xFEED
+
+    class AttributeDefs(CustomCluster.AttributeDefs):
+        """CTM Lyng custom diagnostics cluster attribute definitions."""
+
+        ctm_last_reset_info: Final = foundation.ZCLAttributeDef(
+            id=0x0000,
+            type=t.uint8_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_last_extended_reset_info: Final = foundation.ZCLAttributeDef(
+            id=0x0001,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_reboot_counter: Final = foundation.ZCLAttributeDef(
+            id=0x0002,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_last_hop_lqi: Final = foundation.ZCLAttributeDef(
+            id=0x0003,
+            type=t.uint8_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_last_hop_rssi: Final = foundation.ZCLAttributeDef(
+            id=0x0004,
+            type=t.int8s,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_tx_power: Final = foundation.ZCLAttributeDef(
+            id=0x0005,
+            type=t.int8s,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_parent_node_id: Final = foundation.ZCLAttributeDef(
+            id=0x0006,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_button_0_click_counter: Final = foundation.ZCLAttributeDef(
+            id=0x0010,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_button_0_ms_click_counter: Final = foundation.ZCLAttributeDef(
+            id=0x0020,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_debug_int: Final = foundation.ZCLAttributeDef(
+            id=0x0401,
+            type=t.uint32_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        ctm_cluster_revision: Final = foundation.ZCLAttributeDef(
+            id=0xFFFD,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+
+
+class CTMStoveGuardCluster(CustomCluster):
+    """CTM Lyng custom stove guard cluster."""
+
+    name = "CtmStoveGuard"
+    cluster_id = 0xFFC9
+
+    class AttributeDefs(CustomCluster.AttributeDefs):
+        """CTM Lyng stove guard cluster attribute definitions."""
+
+        ctm_alarm_status: Final = foundation.ZCLAttributeDef(
+            id=0x0001,
+            type=AlarmStatus,
```

Comment:
For these devices, set `zcl_type=foundation.DataTypeId.uint8` and then you can have `AlarmStatus` be an `enum8` subclass.

For the vast majority of devices the ZCL type can be inferred from the Python type but there are many devices that do not follow the specification and use bare integers instead of enums.